src/Model/SnapshotChildrenCollection.php

```diff
@@ -19,6 +19,8 @@
  * @extends AbstractLazyCollection<array-key, PageInterface>
  *
  * @author Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * @psalm-suppress PropertyNotSetInConstructor
```

Comment:
Can you add an explanation about the why ?

And we should do a PR for fixing the phpdoc of https://github.com/doctrine/collections/blob/1.6.x/lib/Doctrine/Common/Collections/AbstractLazyCollection.php

Do you want to do it or should I ?

Reply:
If you have time to do it, it would be nice. IMHO in this case I see nothing wrong with that class. Psalm complains about a property that is not set on the construct by design, because it is a lazy collection.

Reply:
It should be
```
/**
     * The backed collection to use
     *
     * @psalm-var Collection<TKey,T>|null
     * @var Collection<mixed>|null
     */
    protected $collection;
```
then

Reply:
Yep, it is more safe that way. But it is not wrong as of now, because all the methods that use collection will initialize it first.

Reply:
Added a comment.